client/components/banner/test/index.jsx

```diff
@@ -1,165 +1,181 @@
+		expect( container.firstChild ).toHaveClass( 'is-jetpack' );
+		expect( container.getElementsByClassName( 'jetpack-logo' ) ).toHaveLength( 1 );
 	} );
 
 	test( 'should render have .is-horizontal class if horizontal prop is defined', () => {
-		const comp = shallow( <Banner { ...props } horizontal /> );
-		expect( comp.find( '.is-horizontal' ) ).toHaveLength( 1 );
+		const { container } = render( <Banner { ...props } horizontal /> );
+		expect( container.firstChild ).toHaveClass( 'is-horizontal' );
 	} );
 
 	test( 'should render a <PlanPrice /> when price is specified', () => {
-		const comp = shallow( <Banner { ...props } price={ 100 } /> );
-		expect( comp.find( PlanPrice ) ).toHaveLength( 1 );
+		render( <Banner { ...props } price={ 100 } /> );
+		expect( screen.queryByTestId( 'banner-price-plan-mock' ) ).toHaveTextContent( '100' );
 	} );
 
 	test( 'should render two <PlanPrice /> components when there are two prices', () => {
-		const comp = shallow( <Banner { ...props } price={ [ 100, 80 ] } /> );
-		expect( comp.find( PlanPrice ) ).toHaveLength( 2 );
+		render( <Banner { ...props } price={ [ 100, 80 ] } /> );
+		expect( screen.queryAllByTestId( 'banner-price-plan-mock' ) ).toHaveLength( 2 );
 	} );
 
 	test( 'should render no <PlanPrice /> components when there are no prices', () => {
-		const comp = shallow( <Banner { ...props } /> );
-		expect( comp.find( PlanPrice ) ).toHaveLength( 0 );
+		render( <Banner { ...props } /> );
+		expect( screen.queryByTestId( 'banner-price-plan-mock' ) ).toBeNull();
 	} );
 
 	test( 'should render a .banner__description when description is specified', () => {
-		const comp = shallow( <Banner { ...props } description="test" /> );
-		expect( comp.find( '.banner__description' ) ).toHaveLength( 1 );
+		render( <Banner { ...props } description="test" /> );
+		expect( screen.getByText( 'test' ) ).toHaveClass( 'banner__description' );
 	} );
 
 	test( 'should not render a .banner__description when description is not specified', () => {
-		const comp = shallow( <Banner { ...props } /> );
-		expect( comp.find( '.banner__description' ) ).toHaveLength( 0 );
+		const { container } = render( <Banner { ...props } /> );
+		expect( container.getElementsByClassName( 'banner__description' ) ).toHaveLength( 0 );
 	} );
 
 	test( 'should render a .banner__list when list is specified', () => {
-		const comp = shallow( <Banner { ...props } list={ [ 'test1', 'test2' ] } /> );
-		expect( comp.find( '.banner__list' ) ).toHaveLength( 1 );
-		expect( comp.find( '.banner__list li' ) ).toHaveLength( 2 );
-		expect( comp.find( '.banner__list li' ).at( 0 ).text() ).toContain( 'test1' );
-		expect( comp.find( '.banner__list li' ).at( 1 ).text() ).toContain( 'test2' );
+		render( <Banner { ...props } list={ [ 'test1', 'test2' ] } /> );
+		expect( screen.queryByRole( 'list' ) ).toBeVisible();
+		expect( screen.queryAllByRole( 'listitem' ) ).toHaveLength( 2 );
+		expect( screen.getByText( 'test1' ) ).toBeVisible();
```

Comment:
I think we could improve those to use the result of `screen.queryAllByRole( 'listitem' )` and just perform `.toHaveTextContent()` over the `[ 0 ]` and `[ 1 ]` elements of the result. Because right now, this test would pass even if we render `test1` and `test2` anywhere else in the component (outside if the list items).

Reply:
That is really helpful, thank you! After you gave this feedback for other tests (i.e. the test for `should render a .banner__description when description is specified`), I looked to see if I could do the same here, but there weren't any classes, so I didn't think it was possible. But I need to remember I can use indexes for queries that result in arrays. 💡